### PR TITLE
Add config for distributor layer of smart gateway

### PR DESCRIPTION
**Pull request by mdubbyap**

### CODE REVIEW

File: config/loader.go [hunk 1]
```diff
@@ -211,11 +212,17 @@ func (s *signalFxLoader) Forwarder(conf *ForwardTo) (protocol.Forwarder, error)
 		TraceSample:        conf.TraceSample,
 	}
 	if sfConf.TraceSample != nil {
-		sfConf.TraceSample.EtcdServer = conf.Server
 		sfConf.TraceSample.EtcdClient = conf.Client
 		sfConf.TraceSample.AdditionalDimensions = datapoint.AddMaps(conf.AdditionalDimensions, conf.TraceSample.AdditionalDimensions)
 		sfConf.TraceSample.ClusterName = conf.ClusterName
 	}
+	if conf.TraceDistributor != nil {
```
Comment (mpetazzoni) [inline config/loader.go]: You should protect against both being defined.

File: config/config.go [hunk 2]
```diff
@@ -51,9 +51,9 @@ type ForwardTo struct {
 	DimensionsOrder      []string                    `json:",omitempty"`
 	Filters              *filtering.FilterObj        `json:",omitempty"`
 	TraceSample          *sampling.SmartSampleConfig `json:",omitempty"`
+	TraceDistributor     *sampling.SmartSampleConfig `json:",omitempty"`
```
Comment (mpetazzoni) [inline config/config.go]: I understand why it ends up being done this way, but it means there is probably nothing preventing people from putting sampler config things (like `BackupLocation` or `ClusterPercent`) in their distributor config, which can be very confusing.

File: config/loader.go (hunk 1 above)
Comment (mdubbyap) [reply config/loader.go]: agree

File: config/config.go (hunk 2 above)
Comment (mdubbyap) [reply config/config.go]: Even if i created a separate object, they could put in a key of `BackupLocation` and we would ignore it just as we'll currently ignore it. This solution i have put here makes the backend to the sampler far simpler codewise.

File: config/loader_test.go [hunk 3]
```diff
@@ -44,9 +45,12 @@ func TestForwarders(t *testing.T) {
 			{name: "should load CSV forwarder", forwardTo: &ForwardTo{Type: "csv", Filename: pointer.String("datapoints.csv")}, pass: true, reset: func() error { So(os.Remove("datapoints.csv"), ShouldBeNil); return nil }},
 			{name: "signalfx forwarder should work", forwardTo: &ForwardTo{Type: "signalfx"}, pass: true},
 			{name: "signalfx forwarder with sample should fail due to not smart gateway", forwardTo: &ForwardTo{Type: "signalfx", TraceSample: &sampling.SmartSampleConfig{}}, pass: false},
+			{name: "signalfx forwarder with distributor should fail due to not smart gateway", forwardTo: &ForwardTo{Type: "signalfx", TraceDistributor: &sampling.SmartSampleConfig{}}, pass: false},
+			{name: "signalfx forwarder with distributor and smart gateway are incompatible", forwardTo: &ForwardTo{Type: "signalfx", TraceDistributor: &sampling.SmartSampleConfig{}, TraceSample: &sampling.SmartSampleConfig{}}, pass: false},
 		}
 		for _, test := range tests {
 			Convey(test.name, func() {
+				fmt.Println(test.name)
```
Comment (mpetazzoni) [inline config/loader_test.go]: Not sure you want to keep that around?

Comment (mdubbyap) [reply config/loader_test.go]: ack

**Pull request merged**